src/StoreInjector.ts

```diff
@@ -0,0 +1,114 @@
+import WeakMap from '@dojo/shim/WeakMap';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { w } from '@dojo/widget-core/d';
+import { handleDecorator } from '@dojo/widget-core/decorators/handleDecorator';
+import { beforeProperties } from '@dojo/widget-core/decorators/beforeProperties';
+import { alwaysRender } from '@dojo/widget-core/decorators/alwaysRender';
+import { RegistryLabel, Constructor, DNode } from '@dojo/widget-core/interfaces';
+import { Store } from './Store';
+import { Injector } from '@dojo/widget-core/Injector';
+
+const registeredInjectorsMap: WeakMap<WidgetBase, StoreInjector[]> = new WeakMap();
```

Comment:
Do you think this would be better as `WeakMap<WidgetBase, Set<StoreInjector>>` ?

Reply:
Think it's worthwhile introducing `Set` (plus the shim), what benefit does the `Set` get us?

Reply:
Ok, fair enough. nothing major, was just a thought.